### DISCUSSION TITLE
how can I prefetch infinite query in getServerSideProps in next.js?

### DISCUSSION THREAD

**Discussion by aniket-jena-dev** [TanStack/query#4031] [Q&A]
I am implementing pagination and having trouble to cache the infinite query (idk if it is even possible)
getServerSideProps is as follows...
`
export const getServerSideProps: GetServerSideProps = async (context) => {
  const queryClient = new QueryClient()

  await queryClient.prefetchInfiniteQuery(
    ["posts"],
    async ({ pageParam = "" }) => {
      const res = await fetch(
        `${
          process.env.NEXT_PUBLIC_API
        }/api/posts?limit=${15}&cursor=${pageParam}`
      )
      return await res.json()
    }
  )

  return {
    props: {
      dehydratedState: dehydrate(queryClient),
    },
  }
}
`

**Comment by TkDodo** [comment]
you'd have to be more precise on what the issue is ...
